Sources/DeepLearning/Trainer.swift

```diff
@@ -13,26 +13,18 @@
 // limitations under the License.
 
 /// SupervisedTrainer implements a standard training loop for a model.
-public struct SupervisedTrainer<Model, Opt: Optimizer, Loss: Differentiable>
-    where Opt.Model == Model,
-          Loss: FloatingPoint,
-          Loss == Loss.CotangentVector {
-    typealias LossFn = @differentiable (Model.Output, Model.Output) -> Loss
-
-    var model: Model
-    var optimizer: Opt
-    var lossFn: LossFn
-
-    init(model: Model, with optimizer: Opt, loss: @escaping LossFn) {
-        self.model = model
-        self.optimizer = optimizer
-        self.lossFn = loss
-    }
-
-    mutating func fit(x: Model.Input, y: Model.Output, forSteps numSteps: Int) {
+public struct SupervisedTrainer {
```

Comment:
Define a `private init() {}` so that the default initializer won't be used anywhere. Or make this an `open class` so that this can be subclassed and extended. The `fit` function should be `open` in that case too.

Reply:
Since I don't think it makes sense to override the fit function, I've left it as a public struct, and added a private initializer.